package.json

```diff
@@ -26,9 +26,12 @@
     "react": "^16.13.1",
     "react-datasheet": "^1.4.8",
     "react-dom": "^16.13.1",
+    "react-hotkeys": "^2.0.0",
     "react-redux": "^7.2.1",
     "react-scripts": "^3.4.3",
     "redux": "^4.0.5",
+    "redux-devtools-extension": "^2.13.8",
+    "redux-undo": "^1.0.1",
```

Comment:
Biblioteka, która ma wbudowaną obsługę undo/redo, która przechowuje stan w postaci:
Redux-undo działa w ten sposób, że przechowuje stan w postaci obiektu: 
{ 
 past: [State]
 present: State
 future: [State]
}
Niestety taki sposób jest bardzo kosztowny, ponieważ przechowujemy całe harmonogramy dla każdej zmiany. Zostanie ona zmieniona, prawdopodbnie na Command DP, tak żeby przechowywać tylko akcje, które zostały wykonane.

Reply:
Okazało się że nie jest to aż takie kosztowne, jak mogłoby się wydawać ( coś około 14KB dla jednego obiektu. ) Więc moim zdaniem całkiem może być taka implementacja jak jest teraz.